# coding: utf-8

from __future__ import absolute_import
from datetime import date, datetime  # noqa: F401

from typing import List, Dict  # noqa: F401

from swagger_server.models.base_model_ import Model
from swagger_server.models.error_auth_errors import ErrorAuthErrors  # noqa: F401,E501
from swagger_server import util


class ErrorAuth(Model):
    """NOTE: This class is auto generated by the swagger code generator program.

    Do not edit the class manually.
    """

    def __init__(self, trace: str=None, errors: List[ErrorAuthErrors]=None):  # noqa: E501
        """ErrorAuth - a model defined in Swagger

        :param trace: The trace of this ErrorAuth.  # noqa: E501
        :type trace: str
        :param errors: The errors of this ErrorAuth.  # noqa: E501
        :type errors: List[ErrorAuthErrors]
        """
        self.swagger_types = {
            'trace': str,
            'errors': List[ErrorAuthErrors]
        }

        self.attribute_map = {
            'trace': 'trace',
            'errors': 'errors'
        }

        self._trace = trace
        self._errors = errors

    @classmethod
    def from_dict(cls, dikt) -> 'ErrorAuth':
        """Returns the dict as a model

        :param dikt: A dict.
        :type: dict
        :return: The error_auth of this ErrorAuth.  # noqa: E501
        :rtype: ErrorAuth
        """
        return util.deserialize_model(dikt, cls)

    @property
    def trace(self) -> str:
        """Gets the trace of this ErrorAuth.


        :return: The trace of this ErrorAuth.
        :rtype: str
        """
        return self._trace

    @trace.setter
    def trace(self, trace: str):
        """Sets the trace of this ErrorAuth.


        :param trace: The trace of this ErrorAuth.
        :type trace: str
        """
        if trace is None:
            raise ValueError("Invalid value for `trace`, must not be `None`")  # noqa: E501

        self._trace = trace

    @property
    def errors(self) -> List[ErrorAuthErrors]:
        """Gets the errors of this ErrorAuth.


        :return: The errors of this ErrorAuth.
        :rtype: List[ErrorAuthErrors]
        """
        return self._errors

    @errors.setter
    def errors(self, errors: List[ErrorAuthErrors]):
        """Sets the errors of this ErrorAuth.


        :param errors: The errors of this ErrorAuth.
        :type errors: List[ErrorAuthErrors]
        """
        if errors is None:
            raise ValueError("Invalid value for `errors`, must not be `None`")  # noqa: E501

        self._errors = errors
